crypto/asn1/a_utf8.c

```diff
@@ -73,7 +73,7 @@ int UTF8_getc(const unsigned char *str, int len, unsigned long *val)
         value |= (*p++ & 0x3f) << 12;
         value |= (*p++ & 0x3f) << 6;
         value |= *p++ & 0x3f;
-        if (value < 0x10000)
+        if (value < 0x10000 || value > UNICODE_LIMIT)
```

Comment:
Otherwise 4 byte `\xf4\x90\...` would remain valid
```suggestion
        if (value < 0x10000 || value >= UNICODE_LIMIT)
```